### DISCUSSION TITLE
There is a malfunction in Mojo Playground

### DISCUSSION THREAD

**Discussion by mojo-for-ai** [modular/modular#1396] [General]
![Screenshot_٢٠٢٣١١١٠-٠٨٢١١٨_Chrome](https://github.com/modularml/mojo/assets/152310057/898783fa-2710-4cb6-80f3-f7710bea5ae3)

**Comment by mojo-for-ai** [comment]
.  When I run the code, I only see this red line, and I don't see the actual error. I don't know what's wrong. I've been suffering with this problem for about two months.

**Reply by mojo-for-ai** [reply]
![٢٠٢٣١٢٠٥_٠٩٤٣٣٦](https://github.com/modularml/mojo/assets/152310057/c4a7cdd4-e4dd-470f-8ad7-81002dabd5ea)

**Reply by soraros** [reply]
I think people agree that this is not desirable behaviour. Care to write a bug report or a feature request?

**Comment by rarebreed** [comment]
Try doing `var x = 1`

`let` will make something immutable.  So on the 2nd line when you do `x = 2`, that is invalid, because you defined `x` as being immutable with `let`.

It's unfortunate it doesn't give better error messages yet.  Hopefully they will improve on that

**Reply by mojo-for-ai** [reply]
The matter is not related to let being an immutable variable and var being a mutable variable. There is a problem with the playground itself in that it does not show any error in the code. For example, if you make any mistake in your code, the playground will not show this error. It will show this red bar. Just take it.  Example code below: 

![Screenshot_٢٠٢٣١٢٠٥-١٢٤٠٥٩_Chrome](https://github.com/modularml/mojo/assets/152310057/8ebfa688-aad3-48ce-a35a-87b16295eb31)

**Reply by mojo-for-ai** [reply]
It was supposed to show me an error telling me that you put Float instead of Int, but Playground only showed me this red bar.

![٢٠٢٣١٢٠٥_٠٩٤٣٣٦](https://github.com/modularml/mojo/assets/152310057/a3201ef8-8f75-4ec6-bfca-ca5043c456b2)

**Comment by rarebreed** [comment]
> It was supposed to show me an error telling me that you put Float instead of Int, but Playground only showed me this red bar.
> 
> ![٢٠٢٣١٢٠٥_٠٩٤٣٣٦](https://private-user-images.githubusercontent.com/152310057/288012322-a3201ef8-8f75-4ec6-bfca-ca5043c456b2.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4MDkzNjksIm5iZiI6MTcwMTgwOTA2OSwicGF0aCI6Ii8xNTIzMTAwNTcvMjg4MDEyMzIyLWEzMjAxZWY4LThmNzUtNGVjNi1iZmNhLWNhNTA0M2M0NTZiMi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNVQyMDQ0MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYTE1MGNhZjE5YjEyYWIzZjNjOGFlZTZlYTBjZmZkNmQxMzk2YmU0NzRjNThhMzcxNjcxMTAxMzM5NTY5YzcwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2WvPGHV0PvwwN3MeACfjIf8uvu8yNVH6ZjNr10Oavmc)

I agree that the error information on the playground is not very helpful.  It ideally should show the same error as from `mojo run` or `mojo build`

**Comment by ematejska** [comment]
Agreed that this experience is not very good.  As a workaround, have you tried running the local Jupyter which comes with the downloadable Mojo SDK?  The erorrs work in there fine.
